COPYING

```diff
@@ -146,7 +146,7 @@ Files: src/include/timegm.h
   Copyright (C) Copyright 2010-2011 Vicente J. Botet Escriba
   License: Boost Software License, Version 1.0
 
-Files: src/msg/async/AsyncConnection.cc, src/msg/simple/Pipe.cc (sigpipe suppression)
+Files: src/msg/async/PosixStack.cc, src/msg/simple/Pipe.cc (sigpipe suppression)
   Copyright (C) 2010 Tomash Brechko.  All rights reserved.
   License: GPL3
 
```

Comment:
The headers of these files shows different Copyright(s) and different License !!!

Reply:
> please search for sigpipe suppression related code (search “http” for it)

@tchaikov I see that now. Your changes seems to be good. But when I try to check the License, I found that the blog of Tomash Brechko doesn't exist anymore: http://krokisplace.blogspot.in/2010/02/suppressing-sigpipe-in-library.html

Reply:
Umm. I’m not at a desktop right now but GPL-licensed code in our messenger implementations is deeply concerning, since it’s used by clients and those really need to be linkable...If nobody else does it first I will take a look tomorrow (US time) and see what’s up there.

Reply:
@joscollin http://archive.is/uQ9o if you mind to google it and follow the links.

Reply:
@gregsfortytwo i think the code snippet we are using is (also) licensed under LGPL 3.0, see https://github.com/kroki/XProbes/blob/1447f3d93b6dbf273919af15e59f35cca58fcc23/src/libxprobes.c#L156. i am updating our COPYING,copyright to be in sync with this finding.

Reply:
Yes, that is LGPL. I think we can update the URL(s) in `PosixStack.cc`, as if it is picked from this new URL.

Reply:
@gregsfortytwo agreed, GPL is not compatible with LGPL 2.1, so including GPL licensed code this way actually renders almost all of ceph executables a "derivative work" of it, and turns it into GPL'ed. but i just found that the code to suppress SIGPIPE is licensed under LGPL-3. so strictly speaking, by including that code (not linking against it), a large part of Ceph becomes a derivative work of it, and turns it into LGPL-3 licensed. but that hurts less, as LGPL-3 still offers the link harbor.

Reply:
@tchaikov @gregsfortytwo 
GPL is sort contagious, anything that touches it will be tainted with that same requirement.
So to claim that Ceph is LGPL for linking all code that is required for that needs also be LGPL.
And if any file of that selection of files is LGPL 3.1, I think to whole set will need to be treated as is was LGPL 3.1. I have not looked into detail what difference there are between LGPL 2.1 and 3.1, but probably 2.1 is more flexible. And since I would like to integrate rbd device native in the FreeBSD hypervisor, this perhaps could cause problems.  But might I have to take it up with the FreeBSD board.
Now if it is only these 2 files, then it might be a good thing to just replace the files with another implemtation to is LGPL 2.1, and try to enforce that as a project objective?

Reply:
@wjwithagen a PR would be appreciated.

Reply:
I'm taking care of this already. @wjwithagen will want to make sure I don't break BSD but otherwise this will go away shortly. :)

Reply:
@tchaikov 
8-D That is a fair comment, but currently to busy with $dayjob